code/datums/elements/deployable_item.dm

```diff
@@ -137,7 +137,11 @@
 	attached_item.toggle_deployment_flag()
 
 	user.unset_interaction()
-	user.put_in_hands(attached_item)
+
+	if((get_dist(deployed_machine, user) > 1) || deployed_machine.z != user.z)
+		attached_item.forceMove(get_turf(deployed_machine))
```

Comment:
Why can you still redeploy even if you're not on the same z/not next to it? should it just be disallowed in the first place

Reply:
Because the dispenser does this snowflake 4 second delay disassemble. So you have to be next to it to trigger it, but you can just walk off while it's transforming back into a bag.

Reply:
Why not fix the delay not canceling instead

Reply:
There's nothing wrong with the delay though, and it's not cancelling, its just making it drop on the floor.
Basically with a dispenser you're pressing a button, and then it disassembles itself (hence the animation). deployable code wasn't made with that in mind though.

Reply:
uh ok